sklearn_xarray/preprocessing.py

```diff
@@ -1593,3 +1613,134 @@ def reduce(X, return_estimator=False, **fit_params):
         return Xt, estimator
     else:
         return Xt
+
+    
+ 
+class Stackerizer(BaseTransformer):
+
+    """ Transformer to handle higher dimensional data, for instance data
+        sampled in time and location ('x','y','time'), that must be stacked
+        before running Featurizer, and unstacked after prediction.
+
+    Parameters
+    ----------
+    sample_dim : list, tuple
+        List (tuple) of the dimensions used to define how the data is sampled. 
+    
+        If your sample dim has multiple dimensions, for instance x,y,time 
+        these can be passed as a list or tuple. Before stacking, a new 
+        multiindex 'sample' will be created for these dimensions. 
+
+    direction : str, optional
+        "stack" or "unstack" defines the direction of transformation. 
+        Default is "stack"
+    
+    sample_dim : str
+        Name of multiindex used to stack sample dims. Defaults to "sample"
+    
+    transposed : bool
+        Should the output be transposed after stacking. Default is True.
+        
+ 
+    Returns
+    -------
+    Xt : xarray DataArray or Dataset
+        The transformed data.
+    
+    """
+
+    def __init__(
+        self,
+        stack_dims=None,
+        direction="stack",
+        sample_dim ="sample",
```

Comment:
I think it might make sense to make this a little more flexible, the multi-index dim doesn't necessarily have to be the sample dim:
```suggestion
        new_dim="sample",
```